### ISSUE TITLE
Serde support?

### ISSUE THREAD

**Issue by jacobsvante** [bikeshedder/sunspec#2]
Would be nice to be able to serialize the models. E.g. for displaying purposes.

**Comment by bikeshedder** [comment]
Yes. That makes an awful lot of sense. I'll add a `serde` feature. :+1:

**Comment by jacobsvante** [comment]
Sweet! 😀

**Comment by bikeshedder** [comment]
I just added serialization support and was wondering if I should use the original Sunspec identifiers for the JSON format or stick with the "rustified" names?

### Sunspec

```json
{
    "AID": "...",
    "StVnd": "...",
    "Evt": "GROUND_FAULT | TEST_FAILED",
    "Ctl": "AUTOMATIC",
}
```

### Rust

```json
{
    "aid": "...",
    "st_vnd": "...",
    "evt": "GroundFault | TestFailed",
    "ctl": "Automatic",
}
```

Personally I prefer the rust style as it's much easier on the eyes and it's just the "default" serialization of an already rustified model definition.

Do you have actual plans for using this?

**Comment by bikeshedder** [comment]
The bitflags format works in both directions. It's just that it's stringly typed inside the JSON and parsing it with a different language will be a bit annoying. As there are some vendor specific things inside those flags I think it's best to just use it like that and accept the default serialization format bitflags uses.

**Comment by jacobsvante** [comment]
Okay good! I agree with you that it's not optimal. Feels a bit dirty. But since that's bitflags default approach I think it's best to follow that approach.

**Comment by bikeshedder** [comment]
Renaming struct fields and enum variants was a no-brainer. Though the `bitflags` don't seam to support renaming.

In the mean time I'll add serialization with "rustified" names. If anyone needs a JSON format with the actual SunSpec identifiers we need to decide how to serialize `Enum`s and `Bitfield`s. The string format is probably not the best to begin with.

**Comment by jacobsvante** [comment]
Sounds like a good start!

I would love to see a new release with this as well as tokio-modbus 0.11 support!

**Comment by bikeshedder** [comment]
I just released the `sunspec 0.4.0` crate on crates.io:
- https://crates.io/crates/sunspec/0.4.0
